R/mod_packageDependencies.R

```diff
@@ -0,0 +1,336 @@
+#' @keywords internal
+#' 
+packageDependenciesUI <- function(id) {
+  uiOutput(NS(id, 'package_dependencies_ui'))
+}
+
+#' Package Dependencies module's server logic
+#' 
+#' @param id a module id name
+#' @param selected_pkg placeholder
+#' @param user placeholder 
+#' @param changes a reactive value integer count
+#' @param parent the parent (calling module) session information
+#' 
+#' @import dplyr
+#' @importFrom riskmetric pkg_ref pkg_assess pkg_score assess_dependencies assess_reverse_dependencies
+#' @importFrom purrr map_df
+#' @importFrom rlang is_empty
+#' @importFrom glue glue
+#' @importFrom DT renderDataTable formatStyle
+#' @importFrom stringr str_replace word
+#' @importFrom shiny tagList showModal removeModal
+#' @importFrom shinyjs click
+#' @importFrom deepdep deepdep plot_dependencies
+#' @importFrom formattable as.datatable formattable formatter style
+#' @importFrom cli cli_progress_message cli_progress_bar cli_progress_update cli_progress_done
+#' 
+#' @keywords internal
+#' 
+packageDependenciesServer <- function(id, selected_pkg, user, changes, parent) {
+  moduleServer(id, function(input, output, session) {
+    ns <- session$ns
+    
+    metric_wts_df <- dbSelect("SELECT id, name, weight FROM metric")
+    metric_weights <- metric_wts_df$weight
+    names(metric_weights) <- metric_wts_df$name
+    
+    loaded2_db <- reactive({
+      dbSelect('SELECT name FROM package')$name
+    })
+    
+    get_versnScore <- function(pkg_name, id) {
+      
+      riskmetric_assess <-
+        riskmetric::pkg_ref(pkg_name,
+                            source = "pkg_cran_remote",
+                            repos = c("https://cran.rstudio.com")) %>%
+        dplyr::as_tibble() %>%
+        riskmetric::pkg_assess()
+      
+      riskmetric_score <-
+        riskmetric_assess %>%
+        riskmetric::pkg_score(weights = metric_weights)
+      
+      cli::cli_progress_update(id = id)
+      
+      return(list(name = riskmetric_assess$package, version = riskmetric_assess$version, score = round(riskmetric_score$pkg_score,2) ))
+    }
+    
+    # used for adding action buttons to data_table
+    shinyInput <- function(FUN, len, id, ...) {
+      inputs <- character(len)
+      for (i in seq_len(len)) {
+        inputs[i] <- as.character(FUN(paste0(id, i), ...))
+        # change icon from arrow to upload if not loaded into db yet
+        pkg_name <- pkg_df()[i, 3] %>% pull()
+        if (!pkg_name %in% loaded2_db()) {
+          inputs[i] <- gsub("fas fa-arrow-right fa-regular", "fas fa-upload fa-solid", inputs[i])
+        }
+      }
+      inputs
+    }
+    
+    depends <- reactive({
+      req(selected_pkg$name() != "-")
+      riskmetric::pkg_ref(selected_pkg$name()) %>% 
+        riskmetric::assess_dependencies() 
+    })
+    
+    revdeps <- reactive({
+      req(selected_pkg$name() != "-")
+      riskmetric::pkg_ref(selected_pkg$name()) %>% 
+        riskmetric::assess_reverse_dependencies() 
+    })
```

Comment:
Another alternative would be to store the package reference in a reactive so that `riskmetric::assess_dependencies()` and `riskmetric::assess_reverse_dependencies()` is called on that stored reference object.

Reply:
OK, I can create a reactive for the package reference and then have `riskmetric::assess_dependencies()` and 
riskmetric::assess_reverse_dependencies()` refer to that reactive